kernel/rtlil.h

```diff
@@ -150,7 +150,7 @@ namespace RTLIL
 			if (!p[0])
 				return 0;
 
-			log_assert(p[0] == '$' || p[0] == '\\');
+			log_assert(p[0] == '$' || p[0] == '\\' || strncmp(p, "_TECHMAP_", strlen("_TECHMAP_")) == 0);
```

Comment:
Why is this necessary? All IdString-s really should have `$` (private) or `\\` (public).

Reply:
It's necessary in order to use `dict`s instead of `std::map`sIdStrings for one of the changes.  I'm sorry, I forgot which right now; I can find out which modification otherwise causes the crash but you'll have to give me a while.

Reply:
See f3250f43

Reply:
Wait, sorry, I'm confused.  Were the reverted changes around line 786 intentional or not?

Reply:
My original commit (and the several that followed it) reverted your change here: https://github.com/YosysHQ/yosys/commit/f3250f4372cadc53fb7eecafe0596c71cc98881d#diff-f7c403795140175b79df71d870dd8971R572
But the latest force-push doesn't.

Reply:
Oh dear. I see a few more. I've no idea why it would do this since it shouldn't have let me push if I wasn't up to date. Standby for another force push.

Reply:
Gotcha.  OK, no problem.

Reply:
e173d68b

Reply:
In general, static IdString-s should be referenced with `ID::<name>` or `ID(name)` -- the latter being necessary for names beginning with `$`.